src/extractor/scripting_environment_lua.cpp

```diff
@@ -542,48 +648,62 @@ void Sol2ScriptingEnvironment::ProcessElements(
     }
 }
 
-std::vector<std::string> Sol2ScriptingEnvironment::GetNameSuffixList()
+std::vector<std::string>
+Sol2ScriptingEnvironment::GetStringListFromFunction(const std::string &function_name)
 {
     auto &context = GetSol2Context();
     BOOST_ASSERT(context.state.lua_state() != nullptr);
-    std::vector<std::string> suffixes_vector;
-
-    sol::function get_name_suffix_list = context.state["get_name_suffix_list"];
-
-    if (get_name_suffix_list.valid())
+    std::vector<std::string> strings;
+    sol::function function = context.state[function_name];
+    if (function.valid())
     {
-        get_name_suffix_list(suffixes_vector);
+        function(strings);
     }
-
-    return suffixes_vector;
+    return strings;
 }
 
-std::vector<std::string> Sol2ScriptingEnvironment::GetRestrictions()
+std::vector<std::string>
+Sol2ScriptingEnvironment::GetStringListFromTable(const std::string &table_name)
 {
     auto &context = GetSol2Context();
     BOOST_ASSERT(context.state.lua_state() != nullptr);
-    std::vector<std::string> restrictions;
-
-    sol::function get_restrictions = context.state["get_restrictions"];
-
-    if (get_restrictions.valid())
+    std::vector<std::string> strings;
+    sol::table table = context.profile_table[table_name];
+    if (table.valid())
     {
-        get_restrictions(restrictions);
+        for (auto &&pair : table)
+        {
+            strings.push_back(pair.second.as<std::string>());
+        };
     }
-
-    return restrictions;
+    return strings;
 }
 
-void Sol2ScriptingEnvironment::SetupSources()
+std::vector<std::string> Sol2ScriptingEnvironment::GetNameSuffixList()
```

Comment:
This is cool, I think there is somewhere a binding fo `vector:Add` that we now probably only need for v1 and below.